packages/backend-common/config.d.ts

```diff
@@ -127,6 +127,19 @@ export interface Config {
           knexConfig?: object;
         };
       };
+      /**
+       * admin is used to set the parameters for the admin database
+       * This is necessary when using the `ensureExists` option which creates the database if needed.
+       * If this is not set, then the `user` and `password` fields will be used to access the `postgres` database.
+       */
+      admin?: {
```

Comment:
Did you consider making this follow the same structure as the default connection and plugin connections? That way you can likely re-use the same config and logic to support additional settings that "may" be required for connecting to the admin db. For example:

```yaml
backend:
  database:
    connection: connectionString | object
    admin:
      connection:  connectionString | object
    plugins:
       catalog:
         connection: connectionString | object
```

Reply:
In the database connection and plugin connection, the 2 forms accepted are a connection string and an untyped object. I can certainly make `admin` to be an untyped object but it can't be a connection string. This is because, while making the database connection, the previous logic utilises the existing config object, only overriding the database name to `postgres`. Thus in a similar spirit, I am not allowing to override the admin database connection with a connection string, rather just some parameters that overrides those in `connection.database`. This ensures backwards compatibility without making the code ugly IMO.

Let me know how you feel about this.

Still,

Reply:
So, this code has been extended quite a bit since the last time I worked in it.

One option, which I'm personally leaning towards would be to modify [ensureDatabaseExists](https://github.com/backstage/backstage/blob/master/packages/backend-common/src/database/types.ts#L80) to accept an optional `adminDbConfig` as an optional parameter which is passed down from [DatabaseManager.getDatabase](https://github.com/backstage/backstage/blob/master/packages/backend-common/src/database/DatabaseManager.ts#L310). This should hopefully allow you to use the high level functions that operate on the config structure more consistency as suggested. Thoughts on this?

Reply:
Yeah, I spent some time trying to see if I can just modify `ensureDatabaseExists` but it seems like `dbConfig` will only have `client` and `connection` params and hence I can't work with those. I will work from `DatabaseManager` then.

Reply:
I have pushed a new commit which tries to make the changes in the way you are looking for. This commit has added lots of failing tests so I will need to fix them. Not today though as its very late in here. Let me know what you think of the codebase though.